channels/drive/client/drive_file.c

```diff
@@ -570,10 +577,18 @@ BOOL drive_file_set_information(DRIVE_FILE* file, UINT32 FsInformationClass, UIN
 			/* http://msdn.microsoft.com/en-us/library/cc232067.aspx */
 		case FileAllocationInformation:
 			/* http://msdn.microsoft.com/en-us/library/cc232076.aspx */
-			Stream_Read_UINT64(input, size);
+			Stream_Read_UINT32(input, size);
 #ifndef _WIN32
-			if (ftruncate(file->fd, size) != 0)
+			fd = OPEN(file->fullpath, O_RDWR);
```

Comment:
Would be nice if you could move that code to ```winpr/libwinpr/file/file.c``` to implement ```SetFilePointerEx``` for files and remove this platform switch.